spec/01-unit/08-router_spec.lua

```diff
@@ -2067,6 +2067,84 @@ for _, flavor in ipairs({ "traditional", "traditional_compatible", "expressions"
         end)
       end
 
+      describe("check empty route fields", function()
+
+        it("empty methods", function()
+          local use_case = {
+            {
+              service = service,
+              route = {
+                id = "e8fb37f1-102d-461e-9c51-6608a6bb8101",
+                methods = {},
+                paths = { "/foo", },
+                updated_at = 100,
+              },
+            },
+          }
+          assert(new_router(use_case))
```

Comment:
Maybe we can test the result of `get_atc` directly here. We can export `get_atc` as `_get_atc` or something so the unit test can call it directly. It is a stronger test than just the router being built successfully.

Reply:
`get_atc` only returns a string or `nil`, we do not know wether it is right or wrong, we must pass the result to act-router and check the expression.

Reply:
Also `get_atc` has been exported as `_get_atc` for flavor expressions' unit test.

Reply:
We can do a `assert.equal` against a known string, before passing to `new_router`.

Reply:
done